src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -210,6 +229,69 @@ static int ask_password_plymouth(
         return 0;
 }
 
+static void free_consoles(struct console *con, const unsigned int num) {
+        unsigned int n;
+        if (!con || !num)
+                return;
+        for (n = 0; n < num; n++)
+                free(con[n].tty);
+        free(con);
+}
+
+static const char *current_dev = "/dev/console";
+static struct console* collect_consoles(unsigned int * num) {
+        _cleanup_free_ char *active = NULL;
+        const char *word, *state;
+        struct console *con = NULL;
+        size_t con_len = 0, len;
+        int ret;
+
+        assert(num);
+        assert(*num == 0);
```

Comment:
Why this assertion?

We generally have the rule that we don't touch the pointers to return values unless we exit with success (0). Hence, you should keep the count variable local and only set `*num` at the very end of this function.